polkadot/xcm/procedural/tests/builder_pattern.rs

```diff
@@ -100,3 +101,61 @@ fn default_builder_allows_clear_origin_before_buy_execution() {
 		])
 	);
 }
+
+#[test]
+fn bounded_vecs_use_vecs_and_truncate_them() {
+	let claimer = Location::parent();
+	// We can use a vec instead of a bounded vec for specifying hints.
+	let xcm: Xcm<()> = Xcm::builder_unsafe()
+		.set_hints(vec![AssetClaimer { location: claimer.clone() }])
+		.build();
+	assert_eq!(
+		xcm,
+		Xcm(vec![SetHints {
+			hints: BoundedVec::<Hint, HintNumVariants>::truncate_from(vec![AssetClaimer {
+				location: Location::parent()
+			},]),
+		},])
+	);
+
+	// If we include more than the limit they'll get truncated.
```

Comment:
Where do we specify the limit and the way we truncate? 
The test suggests that the last hint is being preserved. Can this behaviour be modified?

This test also showcases unhappy path of having 2+ hints. Could you add a happy path?
This comment might be irrelevant -- see the prdoc comment first.

Reply:
The limit is the number of hints. Truncating is fine since it will never be an issue as long as you don't duplicate hints.